bindings/go/descriptor/descriptor.go

```diff
@@ -0,0 +1,177 @@
+	}
+	return ci, nil
+}
+
+func (c *ComponentIdentity) AsIdentity() runtime.Identity {
+	return runtime.Identity{
+		"name":    c.Name,
+		"version": c.Version,
+	}
+}
+
+type Resource struct {
+	ObjectMeta    `json:",inline"`
+	ExtraIdentity map[string]string `json:"extraIdentity,omitempty"`
+	SourceRefs    []*SourceRef      `json:"sourceRefs,omitempty"`
+	Type          string            `json:"type"`
+	Relation      string            `json:"relation"`
+	Access        Access            `json:"access"`
+	Digest        *Digest           `json:"digest,omitempty"`
+	Size          int64             `json:"size,omitempty"`
+}
+
+func (r *Resource) GetIdentity() map[string]string {
+	m := maps.Clone(r.ExtraIdentity)
+	if m == nil {
+		m = make(map[string]string)
+	}
+	m["name"] = r.Name
+	return m
+}
+
+type Source struct {
+	ObjectMeta    `json:",inline"`
+	ExtraIdentity map[string]string `json:"extraIdentity,omitempty"`
+	Type          string            `json:"type"`
+	Access        Access            `json:"access"`
+}
+
+func (r *Source) GetIdentity() map[string]string {
+	m := maps.Clone(r.ExtraIdentity)
+	if m == nil {
+		m = make(map[string]string)
+	}
+	m["name"] = r.Name
+	return m
+}
+
+type Access struct {
+	runtime.Typed `json:",inline"`
+}
+
+func (a *Access) UnmarshalJSON(data []byte) error {
+	a.Typed = &runtime.Raw{}
+	return json.Unmarshal(data, a.Typed)
+}
+
+func (a *Access) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.Typed)
+}
+
+type Reference struct {
+	Name          string            `json:"name"`
+	ExtraIdentity map[string]string `json:"extraIdentity,omitempty"`
+	Component     string            `json:"componentName"`
+	Version       string            `json:"version"`
+	Digest        Digest            `json:"digest,omitempty"`
+	Labels        []*Label          `json:"labels,omitempty"`
+}
+
+type SourceRef struct {
+	IdentitySelector map[string]string `json:"identitySelector,omitempty"`
+	Labels           map[string]string `json:"labels,omitempty"`
+}
+
+type Meta struct {
+	Version string `json:"schemaVersion"`
+}
+
+type ObjectMeta struct {
+	ComponentIdentity `json:",inline"`
+	Labels            []*Label `json:"labels,omitempty"`
+}
+
+func (o *ObjectMeta) String() string {
+	base := o.Name
+	if o.Version != "" {
+		base += ":" + o.Version
+	}
+	if o.Labels != nil {
+		base += fmt.Sprintf(" (%v)", o.Labels)
+	}
+	return base
+}
+
+type Digest struct {
+	HashAlgorithm          string `json:"hashAlgorithm"`
+	NormalisationAlgorithm string `json:"normalisationAlgorithm"`
+	Value                  string `json:"value"`
+}
+
+type Signature struct {
+	Name      string        `json:"name"`
+	Digest    Digest        `json:"digest"`
+	Signature SignatureSpec `json:"signature"`
+}
+
+type SignatureSpec struct {
+	Algorithm string `json:"algorithm"`
+	Value     string `json:"value"`
+	MediaType string `json:"mediaType"`
+	Issuer    string `json:"issuer,omitempty"`
+}
+
+type Label struct {
```

Comment:
This is missing the signature relevance bool

Reply:
What is that? :O :D

Reply:
A bool indicating if a label should or should not be part of a potential hash for the component version signature

Reply:
See https://github.com/open-component-model/pocm/pull/37/files#diff-d1709647f6bb5514e00c6b06d47ffb39e60f22a1996b5938127df5009f121f15